eng/pipelines/repo-mirror.yml

```diff
@@ -0,0 +1,121 @@
+trigger: none
+pr: none
+
+# Sync every 2 hours
+schedules:
+  - cron: "0 */2 * * *"
+    branches:
+      include:
+      - master
+    always: true
+
+jobs:
+  - job: MirrorRepo
+    pool:
+      vmImage: 'Ubuntu 18.04'
+    variables:
+      skipComponentGovernanceDetection: true
+
+    steps:
+      - checkout: none
+
+      - template: ./templates/steps/repo-mirror.yml
+        parameters:
+          WORKING_DIRECTORY: $(WORKING_DIRECTORY)
+          SOURCE_REPO: $(SOURCE_REPO)
+          SOURCE_BRANCHES: $(SOURCE_BRANCHES)
+          TARGET_ENDPOINTS: $(TARGET_ENDPOINTS)
+
+    strategy:
```

Comment:
I do not think we should do this in a matrix entry as I don't want us to use an agent per repo as that is spreading out more then we need. We should do all this on one job as the amount of time isn't going to hurt.

Reply:
Previously we want a near real time sync every 15min so we want the sync to be as fast as possible. Now seems like this is not the problem. Another concern is that if one agent fail on something we still want the sync keep working, because sometimes clone will timeout as these repos are so big, and sometimes push/clone failed because github randomly failed.

Reply:
We can still ensure one back sync doesn't break others with "continueOnError: true" but that doesn't mean we need to run them on different agents. As the number of repos grow so does the number of agent machines we consume everytime this job runs so I'd like to avoid this job causing too much contention for our machine resources.

As for how often we sync I'm curious what scenarios need real-time syncing if we have some then we can figure out other trigger mechanisms. If needed we can even potentially cause a sync whenever there is a commit in the public repo, but we will need to refactor this solution a little if we want to do that.